compiler-plugin/src/main/java/io/ballerina/stdlib/mcp/plugin/Utils.java

```diff
@@ -97,4 +100,27 @@ public static Optional<AnnotationNode> getToolAnnotationNode(SemanticModel seman
                 )
                 .findFirst();
     }
+
+    public static boolean isMcpServiceFunction(SemanticModel semanticModel,
+                                               FunctionDefinitionNode functionDefinitionNode) {
+        Optional<Symbol> parentSymbol = semanticModel.symbol(functionDefinitionNode.parent());
+
+        if (parentSymbol.isEmpty() || parentSymbol.get().kind() != SymbolKind.SERVICE_DECLARATION) {
```

Comment:
Probably not the expected usage atm, but don't we have to handle service classes also?

Reply:
And service object constructors (which would cover declarations also).